### DISCUSSION TITLE
Persistent volumes [disks] are getting deleted when the kubernetes cluster is deleted

### DISCUSSION THREAD

**Discussion by shreyasarani23** [strimzi/strimzi-kafka-operator#4909] [Q&A] [answered]
Hi I am using azure kubernetes service to deploy my kafka cluster using strimzi. Azure disks will be created when we are using persistent volumes. But the problem is that the azure disks are getting deleted when the aks cluster is deleted because these azure disks are getting under node resource group. So when we delete aks cluster the node resource group will be deleted and all the underlying resources under this node resource group will be deleted. I want to retain the disks even if the aks cluster is deleted So what is the solution?

Please answer my query.

**Comment by ppatierno** [comment, marked as answer]
I don't think it's a question you should ask here, because it's not related Strimzi itself.
Strimzi allows to configure the deletion (or not) of PVs when a Kafka cluster is deleted but not the corresponding Azure disks when the AKS instance is deleted.
You should find more information on the Azure related documentation:

https://docs.microsoft.com/en-us/azure/aks/operator-best-practices-storage

and even here talking about storage classes:

https://docs.microsoft.com/en-us/azure/aks/concepts-storage#storage-classes

It seems to be possible to specify a `reclaimPolicy: Retain` for your purpose.

Anyway it's not Strimzi related.

**Reply by shreyasarani23** [reply]
Configuring reclaimPolicy: Retain will retain the persistent volume only if the kafka pod/cluster is deleted. But doesn't retain when the aks instance is deleted. Do you know to retain the azure disks if the aks instance is deleted?
And does the strimzi support to use persistent volumes with azure files?

**Reply by ppatierno** [reply]
> Do you know to retain the azure disks if the aks instance is deleted?

That's a question for Microsoft Azure support. I should read the doc tbh.

> And does the strimzi support to use persistent volumes with azure files?

Strimzi doesn't take care of the underlying storage technology, it relies on storage class. It asks for a PVCs and it's bounded to a PV that has an underlying storage depending on the used storage class.

**Reply by shreyasarani23** [reply]
Ok got it. Thanks @ppatierno for answering my query.
